addons/sale/models/sale.py

```diff
@@ -202,6 +202,13 @@ def _get_payment_type(self):
         ('confirmation_date_conditional_required', "CHECK( (state IN ('sale', 'done') AND confirmation_date IS NOT NULL) OR state NOT IN ('sale', 'done') )", "A confirmed sales order requires a confirmation date."),
     ]
 
+    @api.multi
+    @api.constrains('company_id')
```

Comment:
Only  `company_id` ?
would be great to check this on `product_it` to ? (along with check If Products are shared or not )